### PR TITLE
[TPC-H] Dask query 17

### PR DESCRIPTION
**Pull request by milesgranger**
Part of https://github.com/coiled/dask-engineering/issues/234

Works with dask.dataframe and dask-expr.

### CODE REVIEW

**Review by fjetter — APPROVED** [review]
Is CI skipped intentionally?

Query looks fine to me (assuming it actually runs)

**Comment by milesgranger** [comment]
Ya, I added the `[skip ci]` since this would skip this test by default. Could at the tpch label, but seemed excessive if we're going to do this for every PR?

File: tests/tpch/test_dask.py [hunk 1]
```diff
@@ -306,3 +306,36 @@ def test_query_7(client, dataset_path, fs):
...
+            from
+                lineitem
+            where
+                l_partkey = p_partkey
+        )
+    """
+    lineitem = dd.read_parquet(dataset_path + "lineitem", filesystem=fs)
+    part = dd.read_parquet(dataset_path + "part", filesystem=fs)
+
+    table = lineitem.merge(part, left_on="l_partkey", right_on="p_partkey", how="left")
```
[the remark this answers is not in the document]
Comment (milesgranger) [reply tests/tpch/test_dask.py:335]: Yep you're right. I made this mistake in other PRs but will go back thru.

[the remark this answers is not in the document]
Comment (milesgranger) [reply tests/tpch/test_dask.py]: Okay, fix this and the others. Thanks for catching it. :)

File: tests/tpch/test_dask.py [hunk 2]
```diff
@@ -519,3 +519,36 @@ def test_query_12(client, dataset_path, fs):
...
+        )
+    """
+    lineitem = dd.read_parquet(dataset_path + "lineitem", filesystem=fs)
+    part = dd.read_parquet(dataset_path + "part", filesystem=fs)
+
+    table = lineitem.merge(part, left_on="l_partkey", right_on="p_partkey", how="inner")
+    table = table[
+        (table.p_brand == "Brand#23")
+        & (table.p_container == "MED BOX")
+        & (table.l_quantity < 0.2 * table.l_quantity.mean())
```
[the remark this answers is not in the document]
Comment (milesgranger) [reply tests/tpch/test_dask.py]: good eye, I agree and have updated. :+1:

File: tests/tpch/test_dask.py [hunk 3]
```diff
@@ -645,3 +645,48 @@ def test_query_14(client, dataset_path, fs):
...
+    """
+    lineitem = dd.read_parquet(dataset_path + "lineitem", filesystem=fs)
+    part = dd.read_parquet(dataset_path + "part", filesystem=fs)
+
+    avg_qnty_by_partkey = (
+        lineitem.groupby("l_partkey")
+        .l_quantity.mean()
+        .to_frame()
+        .rename(columns={"l_quantity": "l_quantity_avg"})
+    )
```
Comment (hendrikmakait) [inline tests/tpch/test_dask.py:679]: This query might be faster if aggregated the joined and pre-filtered DF of lineitems and parts instead. Happy to evaluate that in a follow-up though.

**Review by hendrikmakait — APPROVED** [review]
Thanks, @milesgranger!

**Review by fjetter — APPROVED** [review]

**Pull request merged**